packages/core/strapi/lib/middlewares/public/index.js

```diff
@@ -77,9 +77,10 @@ module.exports = (config, { strapi }) => {
         }),
         config: { auth: false },
       },
+      // All other public GET-routes except /uploads/(.*) which is handled in upload middleware
       {
         method: 'GET',
-        path: '/(.*)',
+        path: '^((?!/uploads/).)*',
```

Comment:
I'm not really good at regex matching, it seems to work well!
I was curious to try this one : `^(?!/uploads/).*`. It seems the same and shorter. Do you think works too?

Reply:
Thanks for the feedback! 
Yeah, I am not a regex-pro either, but without the brackets it still works the same and also needs fewer steps to match. So it should even have more performant in this case. Therefore, will change accordingly!